src/metabase/driver/h2/actions.clj

```diff
@@ -109,3 +109,21 @@
     {:type    error-type
      :message (tru "Some of your values aren’t of the correct type for the database.")
      :errors  {}}))
+
+(defmethod sql-jdbc.actions/maybe-parse-sql-error [:h2 driver-api/violate-permission-constraint]
+  [_driver error-type _database action-type error-message]
+  (when-let [[_match _table-name]
+             (re-find #"Not enough rights for object \"(.+)\"" error-message)]
+    (merge {:type error-type}
+           (case action-type
+             (:table.row/create :model.row/create)
```

Comment:
Idea: `(case (name action-type) "create" ...)`

This way you handle `data-grid.row` as well.

Reply:
I think we should keep this to cover the primitive actions only to reduce confusion since data-grid.row will delegate to table.row/*